### PR TITLE
Suppress stack in VersionConflictEngineException

### PR DESCRIPTION
**Pull request by DaveCTurner**
`VersionConflictEngineException` is thrown on the hot path for updates,
but stack traces are expensive to compute and transport and rarely
useful for this kind of exception. This commit avoids computing the
stack trace for these exceptions.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
@dnhatn the main question is whether these stack traces are ever useful. I've certainly never used one, but I figure you have more experience in this area. I'll link a support case where handling these stack traces appeared a lot in the hot threads.

**Review by dnhatn — APPROVED** [review]
The message of the exception should explain well why an operation is conflicted. LGTM. Thanks, David!

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/2

File: server/src/main/java/org/elasticsearch/index/engine/VersionConflictEngineException.java [hunk 1]
```diff
@@ -57,4 +57,10 @@ public RestStatus status() {
     public VersionConflictEngineException(StreamInput in) throws IOException {
         super(in);
     }
+
+    @Override
+    public synchronized Throwable fillInStackTrace() {
```
Comment (dnhatn) [inline server/src/main/java/org/elasticsearch/index/engine/VersionConflictEngineException.java:62]: @DaveCTurner I think we do not need `synchronized` this method (unlike the super's impl).

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/index/engine/VersionConflictEngineException.java:62]: Ah yes, that was added by IntelliJ. Well spotted. I opened #62555.

**Pull request merged**